tools/osx/crosstool/cc_toolchain_config.bzl

```diff
@@ -347,578 +308,93 @@ def _impl(ctx):
-        cpp_link_dynamic_library_action = action_config(
-            action_name = ACTION_NAMES.cpp_link_dynamic_library,
-            implies = [
-                "contains_objc_source",
-                "has_configured_linker_path",
-                "symbol_counts",
-                "shared_flag",
-                "linkstamps",
-                "output_execpath_flags",
-                "runtime_root_flags",
-                "input_param_flags",
-                "strip_debug_symbols",
-                "linker_param_file",
-                "version_min",
-                "apple_env",
-                "sysroot",
-                "cpp_linker_flags",
-            ],
-            tools = [
-                tool(
-                    path = "cc_wrapper.sh",
-                    execution_requirements = xcode_execution_requirements,
-                ),
-            ],
-        )
-    elif (ctx.attr.cpu == "armeabi-v7a" or
-          ctx.attr.cpu == "darwin_x86_64" or
-          ctx.attr.cpu == "darwin_arm64" or
-          ctx.attr.cpu == "darwin_arm64e" or
-          ctx.attr.cpu == "ios_arm64" or
-          ctx.attr.cpu == "ios_arm64e" or
-          ctx.attr.cpu == "ios_armv7" or
-          ctx.attr.cpu == "ios_i386" or
-          ctx.attr.cpu == "ios_x86_64" or
-          ctx.attr.cpu == "watchos_arm64_32" or
-          ctx.attr.cpu == "watchos_armv7k" or
-          ctx.attr.cpu == "watchos_i386" or
-          ctx.attr.cpu == "watchos_x86_64"):
-        cpp_link_dynamic_library_action = action_config(
-            action_name = ACTION_NAMES.cpp_link_dynamic_library,
-            implies = [
-                "contains_objc_source",
-                "has_configured_linker_path",
-                "symbol_counts",
-                "shared_flag",
-                "linkstamps",
-                "output_execpath_flags",
-                "runtime_root_flags",
-                "input_param_flags",
-                "strip_debug_symbols",
-                "linker_param_file",
-                "version_min",
-                "apple_env",
-                "sysroot",
-            ],
-            tools = [
-                tool(
-                    path = "cc_wrapper.sh",
-                    execution_requirements = xcode_execution_requirements,
-                ),
-            ],
-        )
-    else:
-        cpp_link_dynamic_library_action = None
+    cpp_link_dynamic_library_action = action_config(
+        action_name = ACTION_NAMES.cpp_link_dynamic_library,
+        implies = [
+            "contains_objc_source",
+            "has_configured_linker_path",
+            "symbol_counts",
+            "shared_flag",
+            "linkstamps",
+            "output_execpath_flags",
+            "runtime_root_flags",
+            "input_param_flags",
+            "strip_debug_symbols",
+            "linker_param_file",
+            "version_min",
+            "apple_env",
+            "sysroot",
+            "cpp_linker_flags",
```

Comment:
This action only differed by having `cpp_linker_flags` in 1 branch and not the other, this action exists for all archs, it's just a no-op for some, so we don't need to if on that in both places